go/DEVELOPER.md

```diff
@@ -238,6 +238,132 @@ go run . -help
 go run . -resultsFile gobenchmarks.json -dataSize "100 1000" -concurrentTasks "10 100" -clients all -host localhost -port 6379 -clientCount "1 5" -tls
 ```
 
+### Naming Conventions
+
+#### Function names
+
+For every command that you implement, please use PascalCase.
+
+Examples of good command function names:
+
+- `BZPopMin`
+- `ZRem`
+- `PExpireWithOptions`
+- `SetWithOptions`
+- `Decr`
+
+Examples of bad command function names:
+
+- `zRange`
+- `hincrbyfloat`
+- `Sdiffstore`
+
+#### Function names for examples
+
+`pkgsite` is a program which generates documentation for Go projects.  To install and run it, execute the following:
+
+```bash
+# In the valkey-glide directory
+cd go
+go install golang.org/x/pkgsite/cmd/pkgsite@latest
+pkgsite -open .
+```
+
+In order for `pkgsite` to work properly, examples for API must be written in a very specific format. They should be located in one of the `*_test.go` files in `./api/` and follow this pattern: `Example<ClientType>_<FunctionName>()`. If we wanted to create an example for the `Get` command in `GlideClient`, we would name define our function as follows:
+
+```go
+func ExampleGlideClient_Get() {
+    // Example code here
+}
+```
+
+In cases where we want to show more than one example, we can add extra endings to the function names:
+
+```go
+func ExampleGlideClient_Get_KeyExists() {
+    // Example code here when the key exists
+}
+
+func ExampleGlideClient_Get_KeyNotExists() {
+    // Example code here when the key does not exist
+}
+```
+
+**Note: `ClientType` and `FunctionName` are CASE-SENSITIVE.**
+
+### Documentation
+
+#### Adding links
+
+When adding links, surround the piece of text using square brackets and then put the link reference at the bottom of the comment block.
+
+For example, this links `valkey.io` with the proper reference:
+
+```go
+// SInter gets the intersection of all the given sets.
+//
+// Note: When in cluster mode, all keys must map to the same hash slot.
+//
+// See [valkey.io] for details.
+//
+// Parameters:
+//
+//	keys - The keys of the sets to intersect.
+//
+// Return value:
+//
+//	A `map[string]struct{}` containing members which are present in all given sets.
+//	If one or more sets do not exist, an empty collection will be returned.
+//
+// [valkey.io]: https://valkey.io/commands/sinter/
+```
+
```

Comment:
This is missing information on how to create links to other types for example [options.SortOptions] would link to the documentation for the `SortOptions` struct in the `options` package.